runtime/lua/vscode-neovim.lua

```diff
@@ -1,30 +1,34 @@
 local api = require("vscode-neovim.api")
-local defaults = require("vscode-neovim.defaults")
-local cursor = require("vscode-neovim.cursor")
-
-require("vscode-neovim.highlight")
-require("vscode-neovim.filetype")
 
-local M = {}
+local default_optons = require("vscode-neovim.default-options")
+local cursor = require("vscode-neovim.cursor")
+local highlight = require("vscode-neovim.highlight")
+local filetype = require("vscode-neovim.filetype")
+local numbers = require("vscode-neovim.numbers")
+local autocmds = require("vscode-neovim.autocmds")
 
-M.notify = api.notify
-M.call = api.call
-M.call_range = api.call_range
-M.notify_range = api.notify_range
-M.call_range_pos = api.call_range_pos
-M.notify_range_pos = api.notify_range_pos
+default_optons.setup()
+cursor.setup()
+highlight.setup()
+filetype.setup()
+numbers.setup()
+autocmds.setup()
 
-M.setup = function()
-  defaults.setup()
-  cursor.setup()
-end
+local vscode = {
+  -- actions
+  action = api.action,
+  call = api.call,
+  -- hooks
+  on = api.on,
+  -- vscode settings
+  has_config = api.has_config,
+  get_config = api.get_config,
+  update_config = api.update_config,
+  -- notifications
+  notify = api.notify,
```

Comment:
We might need to make an overload that somehow produces an error if people try to use the old `notify` function.

Reply:
This is completely consistent with the format of vim.notify, and users decide whether to replace it or not. This is the same as the existing notification plugin usage in nvim community.

Reply:
There should be few users using the old Lua API. If we consider compatibility for this, our maintenance cost will be much higher compared to the cost for users to execute `:%s/vscode.notify/vscode.action/g`

![image](https://github.com/vscode-neovim/vscode-neovim/assets/47070852/098e8839-9bcc-4d5b-9521-091549a9640a)
![image](https://github.com/vscode-neovim/vscode-neovim/assets/47070852/f47c2a9a-da27-4683-b61d-766680ceeae3)

Reply:
> There should be few users using the old Lua API.

Yeah that api is very new so breakage is acceptable. Just mention it in release notes.